### PR TITLE
Detect IP Webcams [TPM Decision Please on improving or closing?]

### PR DESCRIPTION
**Pull request by KjellKod**
Thanks to LogRhythm's Tom Salmon who contributed with this in our DPA contest. 
- This rule should not be merged until we have PCAPs with this type of traffic. 
- With the missing pcap should also be added a yaml file that describes the expected output.

### CODE REVIEW

File: rules/Flow_DetectWebcamByVendorMac.lua [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+ 
+   if (TARGET_VENDOR_MAC_PREFIX == nil) then
+      -- List of Vendors MAC Address prefix we want to target
+      -- Note: Must be UPPER CASE for matching. More faster.
+      TARGET_VENDOR_MAC_PREFIX = {}
+      TARGET_VENDOR_MAC_PREFIX["30:8C:FB"] = true -- Dropcam aka. Nest Cam
+      TARGET_VENDOR_MAC_PREFIX["00:24:E4"] = true -- Withings
+      TARGET_VENDOR_MAC_PREFIX["00:18:03"] = true -- ArcSoft SimpliCam
+      TARGET_VENDOR_MAC_PREFIX["BC:51:FE"] = true -- Swann
+      TARGET_VENDOR_MAC_PREFIX["00:50:C2"] = true -- Y-Cam (1)
```
Comment (KjellKod) [inline rules/Flow_DetectWebcamByVendorMac.lua]: This mac might not be valid. 
https://www.wireshark.org/tools/oui-lookup.html, and http://www.macvendors.com/ both claims it's IEEE registration (not yet registred).  Is this a typo?

File: rules/Flow_DetectWebcamByVendorMac.lua [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+      TARGET_VENDOR_MAC_PREFIX["8C:AE:89"] = true -- Y-Cam (2)
+      TARGET_VENDOR_MAC_PREFIX["00:1F:54"] = true -- Lorex
+      
+      TARGET_VENDOR_NAME = {}
+      TARGET_VENDOR_NAME["30:8C:FB"] = "Dropcam/Nest Cam"
+      TARGET_VENDOR_NAME["00:24:E4"] = "Withings"
+      TARGET_VENDOR_NAME["00:18:03"] = "ArcSoft SimpliCam"
+      TARGET_VENDOR_NAME["BC:51:FE"] = "Swann"
+      TARGET_VENDOR_NAME["00:50:C2"] = "Y-Cam (1)" --- IEEE?
```
Comment (KjellKod) [inline rules/Flow_DetectWebcamByVendorMac.lua]: Remove Y-Cam(1) if it cannot be validated or updated to a valid entry

File: rules/Flow_DetectWebcamByVendorMac.lua (hunk 1 above)
Comment (KjellKod) [inline rules/Flow_DetectWebcamByVendorMac.lua]: From Tom Salmon

```
Hey Kjell,

Good spot.

I looked up the vendor prefix on the same wireshark OUI lookup and it returns:
·         00:50:C2:F5:E0:00 Y-cam Solutions Ltd
·         8C:AE:89 Y-cam Solutions Ltd

Which if you search for returns both IEEE and Y-Cam.

Looking up just the prefix (00:50:C2) returns a LOAD of stuff which is probably worse.

I’d say remove that and just stick with the 8C prefix for Y-Cam.

```

**Comment by KjellKod** [comment]
@Schatzman we don't have pcaps for this, should we just ignore to test (obviously) rules without pcaps or should we ignore it by being explicit with some field in the yaml file?  
`TESTING=false`?

**Comment by Schatzman** [comment]
@KjellKod Since the verbose option takes very little effort for our users, I think it would be good to add a testing field. We can also check if a rule doesn't have pcaps but then if TESTING=true we will fail it.

**Comment by KjellKod** [comment]
I am rusty with yaml, would we have to update all the files or could we
just add a field to this specific rule's yaml file? (so by default, if it
doesn't exist the field it is interpreted as true?)

**Comment by KjellKod** [comment]
@slashbel
We are not closing this one but we can't merge it until we have PCAPs that verifies that the rule works.

**Comment by KjellKod** [comment]
candidate for closing

**Comment by KjellKod** [comment]
closed due to product scope

**Pull request closed without merging**